docs/config.rst

```diff
@@ -1,11 +1,623 @@
+
+Here is the ``project`` section of NPSP's ``cumulusci.yml`` file:
+
+.. code-block:: yaml
+
+    project:
+        name: Cumulus
+        package:
+            name: Cumulus
+            name_managed: Nonprofit Success Pack
+            namespace: npsp
+            api_version: 48.0
+            install_class: STG_InstallScript
+            uninstall_class: STG_UninstallScript
+
+Currently under ``$project_config.project__package__namespace`` is the value ``npsp``.
+
+
+Reference Task Return Values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tasks can set an internal ``return_value`` on themselves while executing. This allows one task in a flow to reference the ``return_value`` set on another task that executed prior to it.
+
+To reference a return value on a previous task use the following:
+
+.. code-block:: yaml
+
+    ^^prior_task.return_value
+
+In order to know what is available for ``<return_value>``, find the source code for 
+an individual task.
+
+    Example: Examine the definition for the ``upload_beta`` task inside the universal ``cumulusci.yml`` file.
+
+.. code-block:: yaml
+
+    upload_beta:
+            description: Uploads a beta release of the metadata currently in the packaging org
+            class_path: cumulusci.tasks.salesforce.PackageUpload
+            group: Release Operations
+
+In order to see if anything is being set on ``self.return_values``, we need to find where the class ``cumulusci.tasks.salesforce.PackageUpload`` is defined.
+
+A little digging yields that this class is defined in the file `package_upload.py <>`_ and has a method called ``_set_return_values()``. `This method <https://github.com/SFDO-Tooling/CumulusCI/blob/3cad07ac1cecf438aaf087cdeff7b781a1fc74a1/cumulusci/tasks/salesforce/package_upload.py#L165>`_ sets ``self.return_values`` to a dictionary with the following keys: ``verison_number``, ``version_id``, and ``package_id``.
+
+Now look at the ``release_beta`` flow defined in the universal ``cumulusci.yml`` file:
+
+.. code-block:: yaml
+
+   release_beta:
+        description: Upload and release a beta version of the metadata currently in packaging
+        steps:
+            1:
+                task: upload_beta
+                options:
+                    name: Automated beta release
+            2:
+                task: github_release
+                options:
+                    version: ^^upload_beta.version_number
+            3:
+                task: github_release_notes
+                ignore_failure: True  # Attempt to generate release notes but don't fail build
+                options:
+                    link_pr: True
+                    publish: True
+                    tag: ^^github_release.tag_name
+                    include_empty: True
+                    version_id: ^^upload_beta.version_id
+            4:
+                task: github_master_to_feature
+
+This flow references both ``version_id`` and ``version_number`` return values set on the ``upload_beta`` task.
```

Comment:
```suggestion
This flow is an example of how subsequent tasks can reference the return values of a prior task. In this case, the ``github_release`` task uses the ``version_numer`` set by the ``upload_beta`` task as an option value. This is done with the ``^^upload_beta.version_number`` syntax. Similarly, the ``github_release_notes`` task uses the ``version_id`` set by the ``upload_beta`` task as an option value. This is done with the ``^^upload_beta.version_id`` syntax.
```

Reply:
@Br4nd0R Option value, or optional value?

Reply:
lowercase 'o'

Reply:
@Br4nd0R No, is it "option value" or "optional value". :)

Reply:
@jusstinvholt option value. The `github_release` task takes an option named `version_number`, and the value of that option (the actual version number) is being set to the `version_number` that was returned by the `upload_beta` task.